### ISSUE TITLE
[BUG] mutil optimizer param group

### ISSUE THREAD

**Issue by FL77N** [microsoft/DeepSpeed#5063]
hello, when I use zero2 to train model, I find if I split optimizer‘s params to 2 groups
```python
params = {'params': []}
for module_ in model.modules():
        params['params'].extend(
                [p for n, p in list(module_._parameters.items())
                if p is not None and n != 'bias'])

param_count = len(params['params'])
first_half = params['params'][:param_count // 2]
second_half = params['params'][param_count // 2:]

first_half = {'params': first_half}
second_half = {'params': second_half}

optimizer = torch.optim.AdamW(params=[first_half, second_half])
```

and I will get different gradients on the condition that the same input、the same weight and the loss comparing with not doing this
